arch/arc/core/fault.c

```diff
@@ -123,6 +123,208 @@ static u32_t z_check_thread_stack_fail(const u32_t fault_addr, u32_t sp)
 
 #endif
 
+#ifdef CONFIG_ARC_EXCEPTION_DEBUG
+static void dump_vector_name(u32_t vector)
+{
+
+	if (vector >= 0x10 && vector <= 0xFF) {
+		printk("interrupt %u", vector);
+		return;
+	}
+
+	/* Names are exactly as they appear in Designware ARCv2 ISA
+	 * Programmer's reference manual for easy searching
+	 */
+	switch (vector) {
+	case ARC_EV_RESET:
+		printk("Reset");
+		break;
+	case ARC_EV_MEM_ERROR:
+		printk("Memory Error");
+		break;
+	case ARC_EV_INS_ERROR:
+		printk("Instruction Error");
+		break;
+	case ARC_EV_MACHINE_CHECK:
+		printk("EV_MachineCheck");
+		break;
+	case ARC_EV_TLB_MISS_I:
+		printk("EV_TLBMissI");
+		break;
+	case ARC_EV_TLB_MISS_D:
+		printk("EV_TLBMissD");
+		break;
+	case ARC_EV_PROT_V:
+		printk("EV_ProtV");
+		break;
+	case ARC_EV_PRIVILEGE_V:
+		printk("EV_PrivilegeV");
+		break;
+	case ARC_EV_SWI:
+		printk("EV_SWI");
+		break;
+	case ARC_EV_TRAP:
+		printk("EV_Trap");
+		break;
+	case ARC_EV_EXTENSION:
+		printk("EV_Extension");
+		break;
+	case ARC_EV_DIV_ZERO:
+		printk("EV_DivZero");
+		break;
+	case ARC_EV_DC_ERROR:
+		printk("EV_DCError");
+		break;
+	case ARC_EV_MISALIGNED:
+		printk("EV_Misaligned");
+		break;
+	case ARC_EV_VEC_UNIT:
+		printk("EV_VecUnit");
+		break;
+	default:
+		printk("unknown vector");
+		break;
+	}
+}
+
+/* For EV_ProtV, the numbering/semantics of the parameter are consistent across
+ * several codes, although not all combination will be reported.
```

Comment:
Minor: the databook name for "code" is "cause code" (from the exception cause register) -> I suggest to use "cause" instead of "code" as argument name and in the comments.

Reply:
done

Reply:
done